dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java

```diff
@@ -1357,6 +1362,129 @@ public void test_SaveField_GivenGraphQLCompatibleVariable_ShouldSave(
+     */
+    @Test
+    public void test_SaveNewNoIndexedField_ShouldNotAddESMapping()
+            throws DotSecurityException, DotDataException, IOException {
+        final long time = System.currentTimeMillis();
+        final ContentType type = new ContentTypeDataGen().nextPersisted();
+        try {
+            Field field = FieldBuilder.builder(DateField.class)
+                    .name("field" + time)
+                    .variable("field" + time)
+                    .contentTypeId(type.id())
+                    .indexed(false)
+                    .build();
+            field = fieldAPI.save(field, user);
+
+            final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+            final ESMappingAPIImpl mappingAPI = new ESMappingAPIImpl();
+
+            //verify mapping on working index
+            Map<String, Object> mapping = mappingAPI
+                    .getFieldMappingAsMap(indiciesInfo.getWorking(),
+                            (type.variable() + StringPool.PERIOD + field.variable()));
+            assertFalse(UtilMethods.isSet(mapping));
+
+            //verify mapping on live index
+            mapping = mappingAPI
+                    .getFieldMappingAsMap(indiciesInfo.getLive(),
+                            (type.variable() + StringPool.PERIOD + field.variable()));
+            assertFalse(UtilMethods.isSet(mapping));
+        }finally{
+            ContentTypeDataGen.remove(type);
+        }
+    }
+
+    /**
+     * <b>Method to test:</b> {@link FieldAPIImpl#save(Field, User)}
+     * <p>
+     * <b>Given scenario:</b> A new {@link DateField} is saved and marked as `indexed=true`
+     * <p>
+     * <b>Expected result:</b> The field should be saved and mapped in ES with `type=date`
+     */
+    @Test
+    public void test_SaveNewIndexedField_ShouldAddESMapping()
+            throws DotSecurityException, DotDataException, IOException {
+        final long time = System.currentTimeMillis();
+        final ContentType type = new ContentTypeDataGen().nextPersisted();
+        try {
+            Field field = FieldBuilder.builder(DateField.class)
+                    .name("field" + time)
+                    .variable("field" + time)
+                    .contentTypeId(type.id())
+                    .indexed(true)
+                    .build();
+            field = fieldAPI.save(field, user);
+
+            final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+            final ESMappingAPIImpl mappingAPI = new ESMappingAPIImpl();
+
+            //verify mapping on working index
+            Map<String, String> mapping = (Map<String, String>) mappingAPI
+                    .getFieldMappingAsMap(indiciesInfo.getWorking(),
+                            (type.variable() + StringPool.PERIOD + field.variable())
+                                    .toLowerCase()).get(field.variable());
+            assertTrue(UtilMethods.isSet(mapping.get("type")));
```

Comment:
Codacy found an issue: [The String literal "type" appears 8 times in this file; the first occurrence is on line 1,434](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5866177)